File: server/src/main/kotlin/ba/pascal/weissleder/server/model/framework/support/messages/CommandMessage.kt
package ba.pascal.weissleder.server.model.framework.support.messages

import ba.pascal.weissleder.server.model.framework.support.devices.Command
import java.io.Serializable

// The structure of the following class is based to the Unicorn Project
data class CommandMessage(
    val contextId: Long,
    val command: Command<Serializable?>
)

File: gateway/src/models/messages/command_message.py
from dataclasses import dataclass
from typing import List


from dataclasses_json import dataclass_json, LetterCase, Undefined
from src.models.devices.device import Command
from src.models.devices.nodeID import NodeID


# The structure of the following class is based to the Unicorn Project
@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class CommandMessage:
    device_id: NodeID
    commands: List[Command]

